### PR TITLE
Treat sha as a prefix of FullRevisionHash

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Changes the datastore treatment of the `sha` param to be "a prefix of the `FullRevisionHash` field".
This steps towards #49 and also supports 7-char SHAs.

### CODE REVIEW

**Comment by lukebjerring** [comment]
One sec, relaxing the 10-char minimum for the query param handling...

File: webdriver/webapp_server.go [hunk 1]
```diff
@@ -28,7 +29,7 @@ var (
 )
 
 // StaticTestDataRevision is the SHA for the local (static) test run summaries.
-const StaticTestDataRevision = "b952881825"
+const StaticTestDataRevision = "b952881825e7d3974f5c513e13e544d525c0a631"
```
Comment (Hexcles) [inline webdriver/webapp_server.go]: If you change this variable, you'll need to either fix this line:
https://github.com/web-platform-tests/wpt.fyi/blob/c6886c69ba71795f11be1feb71ec4c58185feda2/webdriver/webapp_server.go#L252

or rename the static directory.

File: util/populate_dev_data.go [hunk 2]
```diff
@@ -45,7 +45,7 @@ func main() {
 	staticDataTime, _ := time.Parse(time.RFC3339, "2017-10-18T00:00:00Z")
 
 	// Follow pattern established in run/*.py data collection code.
-	const staticRunSHA = "b952881825"
+	const staticRunSHA = "b952881825e7d3974f5c513e13e544d525c0a631"
```
Comment (Hexcles) [inline util/populate_dev_data.go]: Same.

File: shared/params.go [hunk 3]
```diff
@@ -140,39 +140,22 @@ const MaxCountMaxValue = 500
 // MaxCountMinValue is the minimum allowed value for the max-count param.
 const MaxCountMinValue = 1
 
-// SHARegex is a regex for SHA[0:10] slice of a git hash.
-var SHARegex = regexp.MustCompile("[0-9a-fA-F]{10,40}")
+// SHARegex is a regex for 7 to 40 char prefix of a git hash.
```
Comment (Hexcles) [inline shared/params.go]: Why is the minimum 7? Is it trying to make sure SHA is unique? What'll happen is the prefix isn't unique?

File: shared/datastore.go [hunk 4]
```diff
@@ -181,24 +218,45 @@ func contains(s []string, x string) bool {
 	return false
 }
 
-// Loads any keys for a full string match or a version prefix (Between [version].* and [version].9*)
-func loadKeysForBrowserVersion(ctx context.Context, query *datastore.Query, version string) (result *mapset.Set, err error) {
+// Loads any keys for a revision prefix or full string match
+func loadKeysForRevision(ctx context.Context, query *datastore.Query, sha string) (result []*datastore.Key, err error) {
```
Comment (Hexcles) [inline shared/datastore.go]: Looks like all the call sites convert the return value to `IntID`. Shall we just return the int ID (also for consistency with the other `load*` function)?

File: shared/params.go (hunk 3 above)
Comment (lukebjerring) [reply shared/params.go]: Don't have a great reason, but the shorter we get, the more likely to clash. If it's not unique, it'll just match more runs. Do you prefer a different minimum?

File: shared/datastore.go (hunk 4 above)
Comment (lukebjerring) [reply shared/datastore.go]: Done.

File: util/populate_dev_data.go (hunk 2 above)
Comment (lukebjerring) [reply util/populate_dev_data.go]: Done.

File: webdriver/webapp_server.go (hunk 1 above)
Comment (lukebjerring) [reply webdriver/webapp_server.go]: Done.

**Pull request merged**